### ISSUE TITLE
How can I log each epoch's metric scores to the results.csv file, in addition to or instead of the "best" scores?

### ISSUE THREAD

**Issue by mattroos** [ultralytics/ultralytics#21116]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

How can I log each epoch's metric scores to the results.csv file, in addition to or instead of the "best" (B) epoch's scores?
I'm training detection models. The results.csv file has these metrics:
```
['epoch',
 'time',
 'train/box_loss',
 'train/cls_loss',
 'train/dfl_loss',
 'metrics/precision(B)',
 'metrics/recall(B)',
 'metrics/mAP50(B)',
 'metrics/mAP50-95(B)',
 'val/box_loss',
 'val/cls_loss',
 'val/dfl_loss',
 'lr/pg0',
 'lr/pg1',
 'lr/pg2']
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mattroos, thank you for your interest in Ultralytics 🚀! This is an automated response to let you know that your question has been received and an Ultralytics engineer will assist you soon.

In the meantime, we recommend checking out the [Docs](https://docs.ultralytics.com/) for useful guidance. For custom training ❓ questions like this, please include as much detail as possible, such as any code you’ve tried, your training logs, and an example of your desired results if possible. This helps us understand your workflow and provide more targeted support.

If this is related to a 🐛 bug or missing feature, a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) is very helpful for debugging or evaluation.

You can also connect with the Ultralytics community:
- For real-time chat, join [Discord](https://discord.com/invite/ultralytics) 🎧
- For deeper discussions, visit [Discourse](https://community.ultralytics.com/)
- Or share and learn on our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

To make sure your question or issue isn’t already resolved in the latest release, please upgrade the `ultralytics` package (along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be run in the following up-to-date verified environments (all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for contributing to Ultralytics! An engineer will be with you shortly.

**Comment by Y-T-G** [comment]
Doesn't it already log the metrics of all epochs?

**Comment by mattroos** [comment]
@Y-T-G from what I can tell, it metric scores for every epoch, but those scores are from the best.pt (B) epoch, not from that current/last epochs last.pt checkpoint. If I'm wrong about that, what does the "(B)" indicate in the metric names such as ''metrics/precision(B)'?

**Comment by Y-T-G** [comment]
B stands for box metrics to distinguish it from masks (M) and pose (P) metrics
